apps/photos/src/components/Collections/CollectionShare/emailShare.tsx

```diff
@@ -2,17 +2,49 @@ import SingleInputForm, {
     SingleInputFormProps,
 } from 'components/SingleInputForm';
 import { GalleryContext } from 'pages/gallery';
-import React, { useContext } from 'react';
+import React, { useContext, useState, useEffect } from 'react';
 import { t } from 'i18next';
 import { shareCollection } from 'services/collectionService';
 import { User } from 'types/user';
 import { handleSharingErrors } from 'utils/error/ui';
 import { getData, LS_KEYS } from 'utils/storage/localStorage';
 import { CollectionShareSharees } from './sharees';
+import { getLocalCollections } from 'services/collectionService';
 
 export default function EmailShare({ collection }) {
     const galleryContext = useContext(GalleryContext);
 
+    const [updatedOptionsList, setUpdatedOptionsList] = useState(['']);
+    let updatedList = [];
+    useEffect(() => {
+        const owner_user1: User = getData(LS_KEYS.USER);
+        const collection_list = getLocalCollections();
+        const getUpdatedOptionsList = async () => {
+            const result = await collection_list;
+            const emails = result.flatMap((item) => {
+                const shareeEmails = item.sharees.map((sharee) => sharee.email);
+                if (item.owner.email) {
```

Comment:
what is the case where `item.owner.email` is not present ?

Reply:
if item.owner.email is not present, that means, it's a collection in our ownership, In that case, I tried to return the sharees mail only

Reply:
so here i guess i have made a mistake, If I return both item.owner.email and sharees when owner.email is present , will sharees return those email with which the owner has shared the collection with?

Reply:
but the code is doing something different no?

Reply:
In the case where item.owner.email is present, I guess it makes more sense if I only return  "return [ item.owner.email];", I have a doubt whether current return [...shareeEmails, item.owner.email] also return the list of emails that the owner has shared the album with(that's something we don't need right).